### ISSUE TITLE
batch renormalization for 3D tensors (conv1d outputs)

### ISSUE THREAD

**Issue by adrienchaton** [rarilurelo/batch_renormalization#1]
Hello,
Thank you for sharing your implementation of 1d batch renormalization.
Considering outputs of linear layers eg. nn.Linear(in_size,out_size) it can be placed afterwards as br = BatchRenorm1d(out_size,r_d_func).

However if we now consider outputs of conv1d layers eg. nn.Conv1d(in_c, out_c, kernel_size) of known width eg. x_out = [mini_batch,out_c,width]. Is it correct to modify your implementation as:

if self.affine:
            #self.weight = Parameter(torch.Tensor(num_features))
            #self.bias = Parameter(torch.Tensor(num_features))
            self.weight = Parameter(torch.zeros(num_features))
            self.bias = Parameter(torch.zeros(num_features))

and provide num_features as (out_c,width) which initializes the weight and bias as matrices instead of vectors ?

If that is correct, it is very interesting for my purpose. Thank you for your insight, it might be useful to make the implementation general for 2D and 3D tensors.

**Comment by adrienchaton** [comment]
One thing also that confuses me ..
with affine=True, it builds self.weight/bias as learnable parameters but these are not used in the forward computation .. hence when back-propagating for instance a reconstruction loss in the network that use batch renormalization, the gradients for these weight/bias are None .. is that right ?
